### PR TITLE
Add documentation for CURRENT_CATALOG and CURRENT_SCHEMA

### PR DESCRIPTION
**Pull request by Damans227**
What this PR does / why we need it:
Add documentation for `CURRENT_CATALOG` and `CURRENT_SCHEMA` in **trino/docs/src/main/sphinx/functions/session.rst**

Which issue(s) this PR fixes:
Add documentation for CURRENT_CATALOG and CURRENT_SCHEMA #7833

Special notes for your reviewer:
Please share feedback

### CODE REVIEW

**Review by mosabua — CHANGES_REQUESTED** [review]
We need to confirm that this actually works. I am running 356 locally and it does not work for me.

Here is what I would conclude from the docs and what I observe:

current user works:

```
select current_user;
  _col0
---------
 manfred
(1 row)
```

But the new ones don't.

```
use tpch.tiny;
USE
trino:tiny> select current_catalog;
Query 20210519_170639_00028_memgm failed: line 1:8: Column 'current_catalog' cannot be resolved
select current_catalog

trino:tiny> select CURRENT_CATALOG;
Query 20210519_170653_00029_memgm failed: line 1:8: Column 'current_catalog' cannot be resolved
select CURRENT_CATALOG

trino:tiny> select current_catalog();
Query 20210519_170716_00030_memgm failed: line 1:8: Function 'current_catalog' not registered
select current_catalog()
```

Any advice @martint ?

**Comment by martint** [comment]
This feature is not in 356. It will be released in 357: https://github.com/trinodb/trino/issues/7824

**Comment by mosabua** [comment]
Ah.. thanks @martint .. not sure how I missed the context. Imho this is good to go although maybe we should add how the current catalog and schema are set by adding a link to the ``use`` command .. wdyt?

File: docs/src/main/sphinx/functions/session.rst [hunk 1]
```diff
@@ -8,4 +8,12 @@ Functions providing information about the query execution environment.
 
     Returns the current user running the query.
 
+.. data:: current_catalog
+
+    Returns a character string that represents the current default catalog name.
```
Comment (m57lyra) [inline docs/src/main/sphinx/functions/session.rst:13]: remove the word "default"

File: docs/src/main/sphinx/functions/session.rst [hunk 2]
```diff
@@ -8,4 +8,12 @@ Functions providing information about the query execution environment.
...
+
+.. data:: current_schema
+
+    Returns a character string that represents the current default unqualified schema name.
```
Comment (m57lyra) [inline docs/src/main/sphinx/functions/session.rst:17]: Remove the word "default"

File: docs/src/main/sphinx/functions/session.rst (hunk 1 above)
Comment (m57lyra) [reply docs/src/main/sphinx/functions/session.rst:13]: ("current default" would mean that which is currently set as a default somewhere, which could be different that the current catalog. If current_catalog means what is set somewhere as a default, but you are looking at another catalog presently, then it is mis-named and should be called "default_catalog".

**Review by rosewms — APPROVED** [review]
After the changes @m57lyra suggested are implemented, this should be good to go.

**Comment by electrum** [comment]
I confirmed that these return NULL if the current catalog or schema is not set.

**Comment by mosabua** [comment]
We discussed this. It would be great to add a mention that the USE command sets the current catalog and schema. In the USE command docs we should link to the function docs as well. 

Also note that the JDBC and CLI have parameters to set these.

File: docs/src/main/sphinx/functions/session.rst (hunk 1 above)
Comment (Damans227) [reply docs/src/main/sphinx/functions/session.rst]: Thanks for reviewing the PR and suggesting new changes. I have removed the word "default".

File: docs/src/main/sphinx/functions/session.rst (hunk 2 above)
Comment (Damans227) [reply docs/src/main/sphinx/functions/session.rst]: Removed "default" from the definition of current_schema parameter.

**Review by Ordinant — COMMENTED** [review]
Using Trino 357 and CLI 357, I see these working:
- select current_catalog;
- select current_schema;

However, I noticed that trying to use FROM out of habit just appends the argument of FROM to the name of the current catalog.schema, then complains that it doesn't exist. People are going to do this. Could the error message be made clearer, stating simply that the FROM keyword is not supported or needed with CURRENT_CATALOG or CURRENT_SCHEMA?

```
trino:sf10000> select current_schema from tpcds;
Query 20210526_203609_00026_kjcm5 failed: line 1:28: Table 'tpch.sf10000.tpcds' does not exist
select current_schema from tpcds

trino:sf10000> select current_schema from zyz;
Query 20210526_203630_00027_kjcm5 failed: line 1:28: Table 'tpch.sf10000.zyz' does not exist
select current_schema from zyz

trino:sf10000> select current_schema from;
Query 20210526_203636_00028_kjcm5 failed: line 1:27: mismatched input '<EOF>'. Expecting: '(', 'LATERAL', 'UNNEST', <identifier>
select current_schema from <EOF>

trino:sf10000> select current_schema from bluesky;
Query 20210526_203646_00029_kjcm5 failed: line 1:28: Table 'tpch.sf10000.bluesky' does not exist
select current_schema from bluesky
```

**Comment by mosabua** [comment]
Could you add the links to the USE command @Damans227

**Comment by Damans227** [comment]
> Could you add the links to the USE command @Damans227

Ok, should I add this as a new note ? Like the one that is already there: 

https://trino.io/docs/current/functions/session.html

**Comment by martint** [comment]
> Could the error message be made clearer, stating simply that the FROM keyword is not supported or needed with CURRENT_CATALOG or CURRENT_SCHEMA?

It is supported. `CURRENT_CATALOG` and `CURRENT_SCHEMA` can appear anywhere an expression or function call appears.

For example:

```
trino:tiny> select * from information_schema.tables where table_catalog = CURRENT_CATALOG and table_schema = CURRENT_SCHEMA;
 table_catalog | table_schema | table_name | table_type
---------------+--------------+------------+------------
 tpch          | tiny         | customer   | BASE TABLE
 tpch          | tiny         | orders     | BASE TABLE
 tpch          | tiny         | lineitem   | BASE TABLE
 tpch          | tiny         | part       | BASE TABLE
 tpch          | tiny         | partsupp   | BASE TABLE
 tpch          | tiny         | supplier   | BASE TABLE
 tpch          | tiny         | nation     | BASE TABLE
 tpch          | tiny         | region     | BASE TABLE
(8 rows)
```

**Comment by electrum** [comment]
Thanks!

**Pull request merged**